client/src/components/Tool/ToolForm.vue

```diff
@@ -0,0 +1,300 @@
+                        type="boolean"
+                        value="false"
+                        @onChange="onChangeRemap"
+                    />
+                    <FormElement
+                        v-if="reuseAllowed"
+                        id="use_cached_job"
+                        title="Attempt to re-use jobs with identical parameters?"
+                        help="This may skip executing jobs that you have already run."
+                        type="boolean"
+                        value="false"
+                        @onChange="onChangeReuse"
+                    />
+                </template>
+                <template v-slot:buttons>
+                    <ButtonSpinner id="execute" title="Execute" @onClick="onExecute" :wait="showExecuting" />
+                </template>
+            </ToolCard>
+        </div>
+    </CurrentUser>
+</template>
+
+<script>
+import Scroller from "vue-scrollto";
+import { getAppRoot } from "onload/loadConfig";
+import { getGalaxyInstance } from "app";
+import { getTool, submitJob, updateTool } from "./services";
+import { send } from "./utilities";
+import ToolCard from "./ToolCard";
+import ButtonSpinner from "components/Common/ButtonSpinner";
+import CurrentUser from "components/providers/CurrentUser";
+import LoadingSpan from "components/LoadingSpan";
+import Form from "components/Form/Form";
+import FormElement from "components/Form/FormElement";
+import ToolSuccess from "./ToolSuccess";
+import Webhook from "components/Common/Webhook";
+
+export default {
+    components: {
+        ButtonSpinner,
+        CurrentUser,
+        LoadingSpan,
+        Form,
+        ToolCard,
+        FormElement,
+        ToolSuccess,
+        Webhook,
+    },
+    props: {
+        id: {
+            type: String,
+            default: null,
+        },
+        version: {
+            type: String,
+            default: null,
+        },
+        job_id: {
+            type: String,
+            default: null,
+        },
+    },
+    data() {
+        return {
+            showLoading: true,
+            showForm: false,
+            showEntryPoints: false,
+            showRecommendation: false,
+            showSuccess: false,
+            showError: false,
+            showExecuting: false,
+            error: null,
+            formConfig: {},
+            formData: {},
+            errorTitle: "",
+            errorContent: null,
+            messageVariant: "",
+            messageText: "",
+            reuse: false,
+            email: false,
+            remap: false,
+            entryPoints: [],
+            jobDef: {},
+            jobResponse: {},
+            validationInternal: null,
+            validationScrollTo: null,
+            validationErrors: null,
+            currentVersion: this.version,
+        };
+    },
+    created() {
+        this.requestTool();
+        const Galaxy = getGalaxyInstance();
+        if (Galaxy.currHistoryPanel) {
```

Comment:
I made a comment in the main thread, but my suggestion is to take ToolForm and separate out its global dependencies in a wrapper component then put ToolForm inside. But yes, currentHistory, config, user, and any other one-time lookup source data for this component should come either from Vuex directly or (more safely) from the providers. The providers ensure the values you want will be initialized before you try to use them.

Reply:
I embedded the history provider but the history refresh should remain as is until the beta history is activated by default.

Reply:
Can you explain why ? You're passing in the history id, which should be synched, regardless of whether you're using the old or new history panel.

Reply:
Right, the `history id` comes from the new provider and works fine, but we are currently making additional manual calls to refresh the history through the `Galaxy` object. We also listen directly to changes in the current history, both of these functions have been transferred from the legacy code. Once the new history is switched on by default we can remove these insertions from the `ToolForm` component.

Reply:
But the history provider should always have to correct current history id, why do we need to listen to history panel twice ? Or does that not work ?

Reply:
1) After submitting a job, we trigger a refresh of the history panel. This is required to quickly show the new datasets in the history panel.
2) When a dataset is added to the legacy history, the tool form needs to request an update in order to display the new datasets instantly in the data selector.

Reply:
Alright, but don't we need to continue doing that when we switch to the beta history ? Sounds like the tool form would still need to be rebuilt, and we'd still like to reset any history refresh backoff ? IOW it might be a good idea to think about this now ?

Reply:
Once the new history is active, we can listen to its state changes by listening to regular events emitted from the corresponding component or provider.

Reply:
We can move forward, but what you're describing are events that the history cannot know about, it will not emit events when a job is submitted, since it has no idea that was done. That's an event that either the form needs to emit and that the provider listens to (or the provider provides a refresh callback ?)

Reply:
I am describing the currently existing framework and the two events which made it through our vuefication process. I would imagine that once the job is submitted, a web socket? tells the history to update. Otherwise we could just remove the update request and accept a short delay until the continuous history request updates by itself, either way my emphasis is that this is not really related to this PR.

Reply:
I'm not sure it's a good idea to ignore these questions while modernizing a component, but I trust your plan, so let's go with it.